### ISSUE TITLE
Showing dates/years on timeline scrollbar

### ISSUE THREAD

**Issue by dlauper** [nextcloud/photos#426]
**Is your feature request related to a problem? Please describe.**
My photo folder contains 43688 pictures. Going back more than a couple of hundred pictures is not really doable, since it would take a day to go back that far. 

**Describe the solution you'd like**
I propose a real timeline as or next to the scrolling bar. Kind of like [Plex](https://www.plex.tv/wp-content/uploads/2018/05/image-photos-videos-timeline-1440x1320.jpg)  but using the full height for spreading out the dates or like Google Photos and probably some others as well,

Cherry on top: if you drag the right-hand side vertical scrollbar to the left, it zooms in allowing for a more precise positioning. Or maybe when clicking it holding ctrl it zooms in to the range of a year. 

**Describe alternatives you've considered**
Album view. Which makes me never ever use the timeline. 

**Additional context**
I'm aware that this could be challenging given the way pictures are loaded step by step. I assume that somehow the total length of the page would need to be known beforhand. The upside is, that this concept could be applied to everything, not just dates, but alphabetical and size sorting too. 

This feature request is part of the big picture of my Photos vision discussion at the [forum](https://help.nextcloud.com/t/ideas-for-revolutionary-photo-organization/89278).

**Comment by skjnldsv** [comment]
Similar: #45

**Comment by dlauper** [comment]
> 
> 
> Similar: #45

Kind of, but #45 doesn't allow you to to go back to February 2008 for instance.

**Comment by mattis75** [comment]
I value the photo app and would also appreciate a possibility to view and scroll between dates in the slideshow view. This would be great and make the app far more usable. regards. matthias

**Comment by Mikescops** [comment]
This is a very complex feature given the Dav backend, we would have to pre-process all existing image in the background to create "groups" of photos by date to be able to know how much data we have per day/month/year.

**Comment by skjnldsv** [comment]
> This is a very complex feature given the Dav backend, we would have to pre-process all existing image in the background to create "groups" of photos by date to be able to know how much data we have per day/month/year.

Unless we take just pre-defined ranges. Like for the last year we take every month and then every year
Then if you click we just show the closest :shrug:

**Comment by Mikescops** [comment]
Could be completely irelevant if we don't have an idea of what the actual data is.

**Comment by da-anda** [comment]
do frontend applications have to use the DAV backends? Just curious if there is some rule for this (I am new to NC)

**Comment by skjnldsv** [comment]
> do frontend applications have to use the DAV backends? Just curious if there is some rule for this (I am new to NC)

The dav backend is already there, so it's simpler if we all use this.
Otherwise we'd have to implement our own endpoint and that can get complicated. I am afraid we'd end up with too many custom endpoints for apps.

**Comment by darrepac** [comment]
My report has been flaged as duplicated of this report, so I copy it here:
The new timeline view of latest Beta is very nice and permits to have a similar user experience than Google Photos or others.
Yet there are an issue when scrolling: it takes time to load next photos and it makes it impossible to jump from exemple from now to 5 years ago very quickly as it is possible on google photos and others.
May-be other services are generating a global date index in order to quickly jump at the right place (?)

**Comment by Mikescops** [comment]
Yes it's the case, they have an index of pictures by range of dates. For Google Photos this comes also with a pre-computed rendering size so that the scroll feels "natural".
Given Nextcloud backend it's not that easy to compete with that.

Considering loading speed, this is related to your machine and the fact that you have or not pre-rendered thumbnails.
I do agree there is work to invest on jumping to old dates in order to speedup the browsing experience.

**Comment by JamesBelchamber** [comment]
So, for anyone else's reference, [this](https://medium.com/google-design/google-photos-45b714dfbed1) is a good long read on how Google Photos does it's thing (from one of the developers).

I really would like this as a feature so I wanted to knock about some ideas (with a view to possibly contributing, or persuading someone else in my network to 😅). Google Photos is _really_ lovely - but as @Mikescops points out, they have some luxuries Nextcloud does not (or, at least, without added complexity) - as well as all that AdSense revenue 💵.

I wonder if we can approximate this, not with an actual scrollbar but with a calendar (or a meter along a vertical edge that approximates a calendar) - and no attempt to even try and make the meter proportional to the amount of photos on each date. The calendar "length" is calculated with a `SEARCH` request on `getlastmodified` - but _ascending_, giving us the oldest photo in the collection. When part of the calendar is clicked on, it refreshes the page with the standard `SEARCH` request - but for files that were last modified before the selected date. It then allows you to scroll down the timeline as normal.

This solution would have one obvious flaw, which is that you can't scroll up - but honestly, you'd have 90% of what people _actually_ want from this feature (being able to jump to a point in the timeline) without needing any precomputed data.

(Feel free to shoot me down on this one - I might be missing something obvious, I'm just trying to come up with an approximate solution that fits within the limitations of WebDAV)

**Comment by andrzej-r** [comment]
Having a "scrubbable view" mentioned in the article with a preallocated page size for all the photos would be great. At least users would be able to scroll down roughly to a desired area. I think (I may be wrong) that a scrubbable or a paginated view is also a prerequisite for any kind of a timeline or a calendar widget. It would be awkward to implement them with an infinite-scroll view.

I'm perfectly fine with a square pitch of photo images - all that variable aspect ratio stuff the article spends a lot of time explaining is not very useful in practice and complicates the layout considerably. I'd rather trade this feature for simplicity and speed.

**Comment by darrepac** [comment]
> I'm perfectly fine with a square pitch of photo images - all that variable aspect ratio stuff the article spends a lot of time explaining is not very useful in practice and complicates the layout considerably. I'd rather trade this feature for simplicity and speed.

I do agree

**Comment by galandilias** [comment]
How to sponsor this ticket? I am really interested in the implementation and willing to put some money for it.

**Comment by darrepac** [comment]
Hello,

what is the status of this issue? is the implementation planned? thx

**Comment by BlackLight** [comment]
The scrollable timeline would be amazing, but difficult to implement given the DAV constraints.

The calendar selector sounds like a good trade-off and, as mentioned by @JamesBelchamber, it's even relatively trivial to implement with a filter on the last modified date.

Given that this issue has been open for one year and a half, it'd be interesting to know if it has been prioritized - and, if not, what are the objections around a datetime picker.

**Comment by svenb1234** [comment]
> implement with a filter on the last modified date.

For photos this is useless. Is there any other photo app out there which uses modified date instead of date taken?

**Comment by darrepac** [comment]
No news on this topic?

**Comment by nZeus** [comment]
Indeed, any updates on this? This is a pretty important feature for anyone who stores photos

**Comment by invario** [comment]
I check on the progress of this periodically with great interest.

In the meantime, what I've discovered as an acceptable alternate on my Android phone is to use an app called File Manager+.  I add in my server as a Cloud location, change the view settings and end up with this:
![Screenshot_20221015_163102](https://user-images.githubusercontent.com/67800603/196006481-6f2d918b-b57f-46d3-9bd0-d851c07407b4.jpg)

Note this is in a folder with 9820 files. The folder loads VERY fast. And I am able to scroll to any point from top to bottom (sorted reverse Modified Date) instantly and with thumbnails (thanks to Preview Generator)

If File Manager+ can do this, I'm not understanding why apps or the web interface can't.

Granted, this is using modified date, not photo/video EXIF data but surely that's better than nothing...?

**Comment by invario** [comment]
Here's a video demonstrating how fast this can go. This is with the settings for 'details' with large thumbnail.

Uploading Screen_Recording_20221015_165044_File Manager +_1.mp4…

**Comment by invario** [comment]
Here's a video demonstrating how fast this can go. This is with the settings for 'details' with large thumbnail.

https://user-images.githubusercontent.com/67800603/196007583-1e378560-a2c0-462e-a7cd-8316353c5216.mp4

**Comment by invario** [comment]
Here's an app on Android that does exactly what Google photos does. 
https://play.google.com/store/apps/details?id=com.nkming.nc_photos

**Comment by kylew01** [comment]
I too would love to see this implemented. Scrolling through a few hundred thousand photos on NC isn't even practical at all, but it's effortless in Google Photos given you can jump right into a year/month. The current implementation is even less useful on the iPhone app.

**Comment by lbesnard** [comment]
> Here's an app on Android that does exactly what Google photos does.
> https://play.google.com/store/apps/details?id=com.nkming.nc_photos

pretty good stuff

**Comment by keesfluitman** [comment]
> Here's an app on Android that does exactly what Google photos does. https://play.google.com/store/apps/details?id=com.nkming.nc_photos

This is actually exactly what I needed. And also what Nextcloud Photos app needs on browsers. Why is this not integrated within the android app and dashboard? It uses the nextcloud's previews, so has all the potential for a good google photos alternative.

**Comment by galandilias** [comment]
Well,  as the app is paid I assume that it will not be incorporated soon into NC android client... And yes - it is perfectly matching my requirements as well - I would only add notifications as on Google from time to time about the day in the past

**Comment by keesfluitman** [comment]
I think it's normal to pay for good working software imho. Software like Nextcloud (or owncloud) allows me to locally host my own cloud for my family and friends. An app like this, running locally, would be well worth my money. Having all photos on my server, plus a properly configurable photo viewer is worth millions. If only I could configure the android/apple app Photos to connect to my server and use nextcloud's preview files and read the exif data. it's such a shame people don't care about all their data being in the cloud.

**Comment by invario** [comment]
> Well, as the app is paid I assume that it will not be incorporated soon into NC android client... And yes - it is perfectly matching my requirements as well - I would only add notifications as on Google from time to time about the day in the past

It's not a paid app. There is a fully functional ad supported free version which I linked above. I paid for it because that's how much I valued it and because I wanted to encourage and support its developer.

I would gladly do the same for this one if the functionality were ever there but in 2 years I don't recall it ever even working. I have folders containing thousands of photos/videos that is inaccessible in this app because it simply can't process it correctly. I and others have submitted numerous bug reports to no avail and obviously it's not a priority for anyone to look at or fix.

**Comment by keesfluitman** [comment]
> > Well, as the app is paid I assume that it will not be incorporated soon into NC android client... And yes - it is perfectly matching my requirements as well - I would only add notifications as on Google from time to time about the day in the past
> 
> It's not a paid app. There is a fully functional ad supported free version which I linked above. I paid for it because that's how much I valued it and because I wanted to encourage and support its developer.
> 
> I would gladly do the same for this one if the functionality were ever there but in 2 years I don't recall it ever even working. I have folders containing thousands of photos/videos that is inaccessible in this app because it simply can't process it correctly. I and others have submitted numerous bug reports to no avail and obviously it's not a priority for anyone to look at or fix.

Ye, they did explain that it can't be added that easily, but requires a full rewrite. Lots of work...So that's why I guess they stick to this format, untill it's actually worth the investment. Imho, the photos app looks nicer now, but I still can't scroll fast or jump to certain dates.
So still no: Timeline, no Exif....

**Comment by keesfluitman** [comment]
> > Well, as the app is paid I assume that it will not be incorporated soon into NC android client... And yes - it is perfectly matching my requirements as well - I would only add notifications as on Google from time to time about the day in the past
> 
> It's not a paid app. There is a fully functional ad supported free version which I linked above. I paid for it because that's how much I valued it and because I wanted to encourage and support its developer.
> 
> I would gladly do the same for this one if the functionality were ever there but in 2 years I don't recall it ever even working. I have folders containing thousands of photos/videos that is inaccessible in this app because it simply can't process it correctly. I and others have submitted numerous bug reports to no avail and obviously it's not a priority for anyone to look at or fix.

Just as a reminder. I found a great solution with mentioned App + the Nextcloud App Memories. This gives me all I need.

**Comment by invario** [comment]
> > > Well, as the app is paid I assume that it will not be incorporated soon into NC android client... And yes - it is perfectly matching my requirements as well - I would only add notifications as on Google from time to time about the day in the past
> > 
> > 
> > It's not a paid app. There is a fully functional ad supported free version which I linked above. I paid for it because that's how much I valued it and because I wanted to encourage and support its developer.
> > I would gladly do the same for this one if the functionality were ever there but in 2 years I don't recall it ever even working. I have folders containing thousands of photos/videos that is inaccessible in this app because it simply can't process it correctly. I and others have submitted numerous bug reports to no avail and obviously it's not a priority for anyone to look at or fix.
> 
> Just as a reminder. I found a great solution with mentioned App + the Nextcloud App Memories. This gives me all I need.

Memories app:
https://github.com/pulsejet/memories

Just installed the Memories app and in the process of indexing everything. This looks very promising!! Thanks!

**Comment by kylew01** [comment]
I found Memories on Nextcloud to tick all of my checkboxes as well (though it still needs to mature some). I wish Nextcloud would just use that as the default photo experience.

**Comment by JanisPlayer** [comment]
I just looked at how the app works.
And actually, it's implementable without tearing down the entire app.
The app requests data as XML via WebDav (/remote.php/dav/).
All the necessary information is here, such as the date, size, and name.
So, you would simply need to create a timeline with this data (date and count), reserve a point for each day, and only load the images when you reach that point on the timeline.
The images must then be unloaded and replaced when you continue scrolling.
It requires a timeline on the right to navigate through the images.
And on the left, there's just a viewer that loads 20 images at the point where you are on the timeline.
The advantage is that Nextcloud no longer hangs, takes less time to load, and generally requires fewer resources.
However, a similar approach to processing data might also be implemented in the file browser, which has exactly the same problem.
It loads data inefficiently, which only works with hundreds of files, not thousands.
Maybe a developer can write the files that are interesting for such changes.
So maybe another developer can make a PR and improve this app so that it can cope with more images and is usable.

**Comment by Viksie** [comment]
Yet another seeking this invaluable feature for searching/managing an insanely large photo and video collection dating back to 1990...

**Comment by susnux** [comment]
Will be included in nextcloud 35 https://github.com/nextcloud/photos/pull/3496
